src/utils.jl

```diff
@@ -73,17 +73,19 @@ $(TYPEDSIGNATURES)
 
 Return the gradient of `f` at `x`.
 """
-function ctgradient(f::Function, x::ctNumber)
-    return ForwardDiff.derivative(x -> f(x), x)
+function ctgradient(f::Function, x::ctNumber; backend=__auto())
```

Comment:
@gdalle this should be enough to allow a global change of the default + combination of several backends.

NB. @ocots may be sth to do with [Requires.jl](https://github.com/JuliaPackaging/Requires.jl) regarding AD backends.

Reply:
Yes it will work but this `__auto()` switch is not accessible to outside users of the package. The best option would be to let this `backend` kwarg (or variants of it, like `gradient_backend`, `jacobian_backend` flow all the way down from the public API to functions like `ctgradient` and `ctjacobian`.

Reply:
As for Requires, you don't need it. If you decide to switch to e.g. `backend = AutoEnzyme()`, as long as you have the Enzyme package loaded in your session (with `import Enzyme`), it will just work